sdk/python/kfp/dsl/_python_component.py

```diff
@@ -47,3 +50,50 @@ def _python_component(func):
     return func
 
   return _python_component
+
+def component(func):
+  """Decorator for component functions that use ContainerOp.
+  This is useful to enable type checking in the DSL compiler
+
+  Usage:
+  ```python
+  @dsl.component
+  def foobar(model: TFModel(), step: MLStep()):
+    return dsl.ContainerOp()
+  """
+  def _component(*args, **kargs):
+    import inspect
+    fullargspec = inspect.getfullargspec(func)
+    args = fullargspec.args
```

Comment:
This looks confusing to me that the 'args' has the same name as the function input. Are they the same thing?

Reply:
Fixed.